### PR TITLE
Add experimental SYCL Support

### PR DESCRIPTION
**Pull request by G-071**
This PR adds support for running the Kokkos kernels on the SYCL execution space (effectively adding support for running Octo-Tiger on Intel GPUs).

 In order to this, the utilized executors and allocators are changed to their respective SYCL versions when Kokkos SYCL support is detected during compile-time. Unfortunately, to allow the Kokkos kernels to run with SYCL, I also had to change the math functions to use the sycl versions for GPU support (for example use ```sycl::sqrt``` for SYCL instantiation of a kernel).
The PR further adds tests for the SYCL versions of the kernels, as well as CLI arguments for selecting those kernels.

The PR was tested both on a A100 GPU and on a MI100 GPU using the Intel LLVM toolchain (though the MI100 run required some slight Kokkos changes to propagate the architecture flags).

Of course, we need the actual allocators and executors for this to work, so this PR comes with multiple dependencies:
For asynchronous launches with HPX futures, we need to HPX SYCL integration/executors added in STEllAR-GROUP/hpx#6085 as well as the respective hpx-kokkos integration in STEllAR-GROUP/hpx-kokkos#13. We further require the SYCL allocators to instantiate the memory pools (with SYCL buffers) which were added in sycl SC-SGS/cppuddle#15. Lastly, to make Octo-Tiger work with recent (current master) version of HPX, we also require the changes in #428 (most of the commits showing up in this PR are due to 428 not being in master yet -- hence I added this PR only as a draft until then).

For convenience, Octo-Tiger buildscripts with the SYCL toolchain are [available here](https://github.com/STEllAR-GROUP/OctoTigerBuildChain/tree/sycl_toolchain).

### CODE REVIEW

**Comment by G-071** [comment]
@diehlpk I think the Codacy warning is one we can safely ignore! Do you know how to silence Codacy about it, or how to make it skip that test?

File: frontend/init_methods.cpp [hunk 1]
```diff
@@ -118,6 +123,7 @@ void init_executors(void) {
     std::cout << "Initialize executors and masks..." << std::endl;
 #ifdef OCTOTIGER_HAVE_KOKKOS
     if (!Kokkos::is_initialized()) { // gets initialized earlier on root locality
+      // TODO SYCL Need args for distributed build...
```
Comment (diehlpk) [inline frontend/init_methods.cpp:126]: Does that means, we can only run single nodes SYCL?

File: octotiger/common_kernel/kokkos_simd.hpp [hunk 2]
```diff
@@ -165,15 +177,15 @@ template <typename simd_t>
 CUDA_GLOBAL_METHOD inline simd_t asinh_with_serial_fallback(const simd_t input) {
   if constexpr (detail::has_simd_asinh<simd_t>::value) {
     // should consider the SIMD_NAMESPACE for overloads due to argument-dependent name lookup
-    return asinh(input);
+    return sycl_asinh_wrapper(input);
   } else {
     /* static_assert(!std::is_same<simd_t, simd_t>::value, "Using asinh serial fallback!" */
```
Comment (diehlpk) [inline octotiger/common_kernel/kokkos_simd.hpp:182]: Remove?

File: frontend/init_methods.cpp (hunk 1 above)
Comment (G-071) [reply frontend/init_methods.cpp:126]: For SYCL I often had to set the device id at startup using the CLI arguments. However, as far as I know that can also be done using environment variables, or even by just limiting the number of devices per rank in distributed builds. Hence it shouldn't really be a problem for distributed builds.

File: octotiger/common_kernel/kokkos_simd.hpp (hunk 2 above)
Comment (G-071) [reply octotiger/common_kernel/kokkos_simd.hpp:182]: No, the wrapper is required!

**Pull request merged**